### ISSUE TITLE
The bot doesn't claim wished characters

### ISSUE THREAD

**Issue by myde2001** [vivinano/MudaeAutoBot#160]
It happened two weeks ago when I got a character, and it happened now, I noticed and I claimed it myself

[https://i.imgur.com/bQyBVKX.png](https://i.imgur.com/bQyBVKX.png)

[https://i.imgur.com/nXVzZzN.png](https://i.imgur.com/nXVzZzN.png)

**Comment by vivinano** [comment]
Pictures aren't loading or linking to anything for me

**Comment by myde2001** [comment]
Sorry, fixed

**Comment by vivinano** [comment]
Where was your name on the wishes by string ?

**Comment by myde2001** [comment]
> Where was your name on the wishes by string ?

My name is the third one, but it also happened in the past when I got a character much higher that the min_kak I specified and it was wished by another player, it appears that the wish string is messing the bot.

**Comment by vivinano** [comment]
Is there snipe protection in your server ?

**Comment by vivinano** [comment]
Another reason is that it seems like button are enabled so ensure both discum library and my repo is updated as button support was added recently

**Comment by myde2001** [comment]
> 

· Rolls sniping: 0 ($togglesnipe)
· Kakera sniping: 0 ($togglekakerasnipe)

**Comment by myde2001** [comment]
> Another reason is that it seems like button are enabled so ensure both discum library and my repo is updated as button support was added recently

I have discum 1.4.0, I downloaded the last version of the script right now, but I had the previous one at this last one.

Also I modified the code in this line "sniping = settings.get("sniping_enabled",False)" to prevent claiming rolls from other people

**Comment by vivinano** [comment]
it searches for your user id not the "wished by" string if you think it only applies to the wish setting then print out content when finds a wished character

**Comment by vivinano** [comment]
you could also test sniping to true and seeing if that the reason things are disabled

**Comment by myde2001** [comment]
> you could also test sniping to true and seeing if that the reason things are disabled

I had it in true the first time it happened and it didn’t claim it

**Comment by myde2001** [comment]
> it searches for your user id not the "wished by" string if you think it only applies to the wish setting then print out content when finds a wished character

So strange this behavior then

**Comment by borisalmonacid** [comment]
> > it searches for your user id not the "wished by" string if you think it only applies to the wish setting then print out content when finds a wished character
> 
> So strange this behavior then

Hi @myde2001, I have found a strange behaviour, I pointed it out in issue #158.
I think that more than pointing out that it does not make a claim, it is pointing out that something changed in the identifiers of the connection to the channel and server.
What I did to make it work for me was to put the slash_ids as channel_ids, that way the bot is making the connection and therefore the claim.

>    "channel_ids":[
>       channel_ids, slash_ids
>    ],
>    "slash_ids":[
>       slash_ids
>    ],
>    "slash_guild_ids":[
>       slash_ids
>    ],

**Comment by myde2001** [comment]
> 

But I'm not using the slash ids

**Comment by vivinano** [comment]
if you are using buttons then it might be a discum dependecy issue

**Comment by vivinano** [comment]
Closing since haven't recieved a response. 
TLDR:
Possible issue with discum dependency due to buttons ?? 
or possible settings issue
